### PR TITLE
buid system: improve serial port selection

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

- extend `dist/tools/usb-serial/ttys.py` with a `--ifcae-num` parameter
    - the CC2650-Launchpad exposes two USB serials, but only one is the UART bridge. The USB iface number is the only stable difference between the two
- introduce `TTY_BOARD_FILTER` to let boards provide flags for `dist/tools/usb-serial/ttys.py` to filter only matching USB serials
    - if port selection via `MOST_RECENT_PORT=1` is enabled, this will fist apply the filter and select the most recent port of the matching USB serials (if more than one)
- provide `TTY_BOARD_FILTER` for a couple of boards
    - This will allow having a `same54-xpro`, `nrf52840dk`, `nucleo-f767zi`, and a `cc2650-launchpad` plugged in and `make BOARD=<foo> MOST_RECENT_PORT=1 term` will always pick the right USB serial
    - Beware: It cannot tell apart two Atmel/Microchip Xpro boards or two Nulceos or two boards with an embedded J-Link debugger though. So no panacea, but still can avoid manual lookup in many cases

### Testing procedure

Plug in a Nucleo, a `samr21-xpro` (or `same54-xpro`), a random Nucleo, a cc2650-launchpad, and a nRF52840DK. `make BOARD=<foo> MOST_RECENT_PORT=1 term` should always pick the right board, no matter which order the boards were plugged in.

(I tested this successfully :))

### Issues/PRs references

None

### CODE REVIEW

File: dist/tools/usb-serial/ttys.py [hunk 1]
```diff
@@ -52,16 +53,6 @@ def filters_match(filters, tty):
     return True
 
 
-def shorten(string, length):
```
Comment (maribu) [inline dist/tools/usb-serial/ttys.py:55]: This is an unused leftover that I forgot to delete. Maybe I should do so in a separate commit.

Comment (maribu) [reply dist/tools/usb-serial/ttys.py:55]: Done

**Review by benpicco — APPROVED** [review]
Pretty neat - and since this only affects runs with `MOST_RECENT_PORT=1` set it will barely be noticed :wink:

**Comment by maribu** [comment]
Thx!
